### PR TITLE
Adding CDC information as a new page in the AuraDB docs

### PR DESCRIPTION
**Pull request by fiquick**
I would like to make a new page in the AuraDB docs for Change Data Capture (CDC)

This text is not ready for release yet, the Aura team are working on implementing a CDC toggle in the Aura console this week.

I would also like to get some feedback from docs and Elfreda.

### CODE REVIEW

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-323.surge.sh

New pages:
- [Change Data Capture (CDC)](https://neo4j-docs-aura-323.surge.sh/aura/auradb/change-data-capture/overview)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-323.surge.sh

New pages:
- [Change Data Capture (CDC)](https://neo4j-docs-aura-323.surge.sh/aura/auradb/change-data-capture/overview)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-323.surge.sh

New pages:
- [Change Data Capture (CDC)](https://neo4j-docs-aura-323.surge.sh/aura/auradb/change-data-capture/overview)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-323.surge.sh

New pages:
- [Change Data Capture (CDC)](https://neo4j-docs-aura-323.surge.sh/aura/auradb/change-data-capture/overview)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-323.surge.sh

New pages:
- [Change Data Capture (CDC)](https://neo4j-docs-aura-323.surge.sh/aura/auradb/change-data-capture/overview)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-323.surge.sh

New pages:
- [Change Data Capture (CDC)](https://neo4j-docs-aura-323.surge.sh/aura/auradb/change-data-capture/overview)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-323.surge.sh

New pages:
- [Change Data Capture (CDC)](https://neo4j-docs-aura-323.surge.sh/aura/auradb/change-data-capture/overview)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-323.surge.sh

New pages:
- [Change Data Capture (CDC)](https://neo4j-docs-aura-323.surge.sh/aura/auradb/change-data-capture/overview)

**Comment by fiquick** [comment]
Hi @AlexicaWright what do you think of the updated documentation?

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-323.surge.sh

New pages:
- [Change Data Capture (CDC)](https://neo4j-docs-aura-323.surge.sh/aura/auradb/change-data-capture/overview)

File: modules/ROOT/pages/auradb/change-data-capture/overview.adoc [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+Note that default log enrichment mode is `OFF` and change data capture is disabled until it is set as `DIFF` or `FULL`.
+
+Only Admins can change the log enrichment mode. 
+If an Admin changes log enrichment mode from `DIFF` or `FULL` to `OFF` applications dependent on CDC will stop working. For more information see link:https://neo4j.com/docs/cdc/current/getting-started/enrichment-mode/#_disabling_log_enrichment[Neo4j DBMS -> CDC -> Disabling Log Enrichment].
+
+== Performing a restore in Aura with log enrichment mode enabled
+
+If you take a snapshot and restore it to the same instance, or create a new instance from that snapshot, by default log enrichment mode will be set to `OFF`. 
+If the original status of that instance had log enrichment mode set to `DIFF` or `FULL`, the status changes. 
+You need to proactively enable it.
```
Comment (AlexicaWright) [inline modules/ROOT/pages/auradb/change-data-capture/overview.adoc:28]:
```suggestion
To record changes when you take a snapshot and restore it to the same instance, you need to proactively set the log enrichment mode to `DIFF` or `FULL` since it is set to `OFF` by default.
```
I'm a little unsure what you are trying to say here, but this is an attempt to make it a little clearer.

File: modules/ROOT/pages/auradb/change-data-capture/overview.adoc [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+CDC assumes that changes can be represented as a continuous stream of changes, each new change building on top of the state of the previous changes.
```
Comment (AlexicaWright) [inline modules/ROOT/pages/auradb/change-data-capture/overview.adoc:30]:
```suggestion
CDC assumes that changes can be represented as a continuous stream, each new change building on top of the state of the previous changes. 
```

File: modules/ROOT/pages/auradb/change-data-capture/overview.adoc [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+This is currently unsupported for databases enabled with CDC.
+
+== Design considerations for applications with CDC in Aura
+
+When designing applications that query using CDC's procedures, those applications track their position in the stream of changes using a cursor. 
+The cursor is valid for a limited time. For more information see link:{neo4j-docs-base-uri}/cdc/current[Neo4j DBMS -> CDC -> Transaction log retention].
+
+Developers should therefore consider high availability for their application so that continuity is maintained in querying in the event the application fails. 
+For example, use Kubernetes to automatically restart the application or a 3rd party product like Apache Kafka or Confluent.
```
Comment (AlexicaWright) [inline modules/ROOT/pages/auradb/change-data-capture/overview.adoc:44]:
```suggestion
Applications that use CDC's procedures to query track their position in the stream of changes using a cursor.
This cursor is valid only for a limited time, see link:{neo4j-docs-base-uri}/cdc/current[Neo4j DBMS -> CDC -> Transaction log retention] for more information.
Therefore, developers should consider high availability for their applications to ensure that query continuity is maintained in the event of application failure.
For example, Kubernetes can be used to automatically restart the application, but also Apache Kafka or Confluent.
```
(Kubernetes is also a 3rd party)

**Review by AlexicaWright — COMMENTED** [review]
Some more suggestions to consider. :)

**Comment by stefano-ottolenghi** [comment]
There's really _three/four_ info points in this page:
- You can see the an instance's CDC status by clicking on an instance in the Aura Console. (this one is quite obvious -- it's the first place I would look for)
- Only Admins can change the log enrichment mode
- When restoring a snapshot from a database that had CDC enabled, the resulting database has it off.
- You can't pause/resume database on which CDC is enabled.

My take is thus whether such a little amount of information warrants the need of a whole new page, and if we were not better off integrating these 4 statements somewhere directly in CDC docs where they are relevant.

The section `Design considerations for applications with CDC in Aura` has nothing to do with Aura. 

There's also a few obvious statements, such as "If an Admin changes log enrichment mode from DIFF or FULL to OFF applications dependent on CDC will stop working." and "default log enrichment mode is OFF and change data capture is disabled until it is set as DIFF or FULL." (which can be shortened to "CDC is off by default."). When you take those all away, it becomes even more apparent that there's very little informational value in the page.

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-323.surge.sh

New pages:
- [Change Data Capture (CDC)](https://neo4j-docs-aura-323.surge.sh/aura/auradb/change-data-capture/overview)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-323.surge.sh

New pages:
- [Change Data Capture (CDC)](https://neo4j-docs-aura-323.surge.sh/aura/auradb/change-data-capture/overview)

File: modules/ROOT/pages/auradb/change-data-capture/overview.adoc [hunk 4]
```diff
@@ -0,0 +1,44 @@
+[[change-data-capture-overview]]
+= Change Data Capture (CDC)
+:description: This section covers changing transaction log enrichment mode using the Aura Console.
+
+label:AuraDB-Professional[] 
+label:AuraDB-Enterprise[]
+
+Change Data Capture tracks changes to your data by querying the transaction logs for Creates, Updates and Deletes, so that you can update other applications or data sources.
```
[the remark this answers is not in the document]
Comment (fiquick) [reply modules/ROOT/pages/auradb/change-data-capture/overview.adoc:8]: https://neo4j.com/docs/cdc/current/ refers to "Change Data Capture"
https://neo4j.com/docs/cdc/current/getting-started/key-considerations/ uses "change data capture"

File: modules/ROOT/pages/auradb/change-data-capture/overview.adoc [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+For more information see link:{neo4j-docs-base-uri}/cdc/current/getting-started/enrichment-mode/[Neo4j DBMS -> CDC -> Enrichment mode].
+
+== Modifying log enrichment mode from within the Aura Console
+
+View an instance’s current log enrichment mode by clicking on an instance in the Aura Console. 
+
+Note that default log enrichment mode is `OFF` and change data capture is disabled until it is set as `DIFF` or `FULL`.
+
+Only Admins can change the log enrichment mode. 
+If an Admin changes log enrichment mode from `DIFF` or `FULL` to `OFF` applications dependent on CDC will stop working. For more information see link:https://neo4j.com/docs/cdc/current/getting-started/enrichment-mode/#_disabling_log_enrichment[Neo4j DBMS -> CDC -> Disabling Log Enrichment].
```
[the remark this answers is not in the document]
Comment (fiquick) [reply modules/ROOT/pages/auradb/change-data-capture/overview.adoc:25]: Let's drop "the" and make it even more concise

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-323.surge.sh

New pages:
- [Change Data Capture (CDC)](https://neo4j-docs-aura-323.surge.sh/aura/auradb/change-data-capture/overview)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-323.surge.sh

New pages:
- [Change Data Capture (CDC)](https://neo4j-docs-aura-323.surge.sh/aura/auradb/change-data-capture/overview)

File: modules/ROOT/pages/auradb/change-data-capture/overview.adoc (hunk 1 above)
Comment (fiquick) [reply modules/ROOT/pages/auradb/change-data-capture/overview.adoc:32]: @l-heemann could you please take a few screenshots of this flow for us to see how it works please?

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-323.surge.sh

New pages:
- [Change Data Capture (CDC)](https://neo4j-docs-aura-323.surge.sh/aura/auradb/change-data-capture/overview)

File: modules/ROOT/pages/auradb/change-data-capture/overview.adoc (hunk 1 above)
Comment (l-heemann) [reply modules/ROOT/pages/auradb/change-data-capture/overview.adoc:32]: I think screenshots need to wait until the button becomes available in Aura. Maybe somebody from the aura-console team who is working on this feature has access to a dev environment or mockup of what it will look like before we release it to customers. I doubt I'll have access to an aura environment with this button before it exists for customers.

For now I think we should maybe rephrase it to be more of a link to the CDC docs on this.

**Comment by l-heemann** [comment]
@stefano-ottolenghi 

> My take is thus whether such a little amount of information warrants the need of a whole new page, and if we were not better off integrating these 4 statements somewhere directly in CDC docs where they are relevant.

I disagree in that there is no value to having CDC related documentation in the Aura docs. Aura has the benefit and drawback that complicated and interconnected features become available as a simple button press. Features such as backup-restore, pause-resume (which leverages backup-resume) and configuring `txLogEnrichment` all require either the use of neo4j-admin or cypher admin commands in self hosted.

We consider CDC to be a set-and-forget technology, where you spend a couple of days configuring your CDC application that then feeds some other system without the need to be maintained all that much. We are worried a person setting up CDC and **another** person pressing a button in the Aura UI that breaks CDC.

I'm not sure if a separate "Aura considerations" page is the right solution. But Aura docs need to have _something_ that lets the user know about the impact CDC has on Aura and what impact Aura has on CDC.

>     * Only Admins can change the log enrichment mode

I don't think CDC docs should document where in Aura the button is to configure enrichment, I expect this to be documented somewhere in the aura docs.

>     * When restoring a snapshot from a database that had CDC enabled, the resulting database has it off.

Restoring a snapshot is very likely to **break your CDC application**. Please link the CDC docs, I think they are pretty clear on the matter.

>     * You can't pause/resume database on which CDC is enabled.

Because this behaves the same as backup restore under the hood, which is not at all what you would expect it to be doing. Especially when backup-restore breaks your CDC application...

File: modules/ROOT/pages/auradb/change-data-capture/overview.adoc [hunk 6]
```diff
@@ -0,0 +1,47 @@
+[[change-data-capture-overview]]
+= Change Data Capture (CDC)
+:description: This section covers changing transaction log enrichment mode using the Aura Console.
+
+label:AuraDB-Professional[]
```
Comment (l-heemann) [inline modules/ROOT/pages/auradb/change-data-capture/overview.adoc:5]: CDC is not available in Aura Pro. There is a limited subset of customers who have Aura Enterprise databases, but use Aura Pro as a testing environment. For those, support may enable CDC on their Pro instances, if the customer asks for it.

File: modules/ROOT/pages/auradb/change-data-capture/overview.adoc (hunk 4 above)
[the remark this answers is not in the document]
Comment (AlexicaWright) [reply modules/ROOT/pages/auradb/change-data-capture/overview.adoc:8]: No, I meant _Creates, Updates and Deletes_ . Also, there should be an Oxford comma in there.

**Comment by stefano-ottolenghi** [comment]
@l-heemann 

tl;dr -- My question at this point is: are all these pieces of information already in the Aura UI and this docs page is just a reference page, or is this docs page _the place_ where this info is exposed? If the latter, then below my thoughts on why this shouldn't be the case.

----

> We are worried a person setting up CDC and another person pressing a button in the Aura UI that breaks CDC.

That's fair, but then docs is not the right place: **it should be in the UI**. When you press the button in Aura to disable CDC, you should get a one-sentence confirmation dialog that warns you about this. Information should be made available as much as possible in the time and place when and where it's most relevant. 

Putting this in docs has the high likelihood that somebody will disable CDC, apps will break, and that group of people will come complaining to us/have a poor experience. We'll protect ourselves by saying that "it's in the docs". That hardly prevented the bad experience from happening though. 

> Restoring a snapshot is very likely to break your CDC application. Please link the CDC docs, I think they are pretty clear on the matter.

As above, this should be a warning in the UI when you are about to restore a snapshot _and_ have CDC enabled. Warn me if I try to do something dangerous, don't bother me otherwise. 

> You can't pause/resume database on which CDC is enabled.

This should go in the UI as well. Grey out the pause/resume button and add a one-sentence note on hover about CDC.

> > Only Admins can change the log enrichment mode
>>
> I don't think CDC docs should document where in Aura the button is to configure enrichment, I expect this to be documented somewhere in the aura docs.

As above, it should be in UI: grey out the enabled button for non-admins and explain on hover. Then if docs need to contain a screenshot of the button because _it needs to be there_, we'll put it there but we should ensure _nobody_ ever needs to land there. This is the experience I would like to have as a user. 

----

For me, there's two broader points here:

1. Information in the UI is much easier to parse, is available where it's relevant, and doesn't require users to read through a verbose list of points that they're unlikely to be interested in entirely and at once. 
2. Every Aura page about a feature brings on-prem and Aura instances farther apart. CDC is one feature, and I feel like people wanting to use CDC should find all the information they need in the relevant docset. Otherwise the Aura docs constantly refers to CDC docs, and CDC docs needs to refer/mention Aura docs. I find it much cleaner to just massage some of the sentences in CDC docs to accommodate Aura (ex. I find [this section](https://neo4j.com/docs/cdc/current/getting-started/key-considerations/#restore-from-backup) descriptive enough, we just need to drop the word `snapshot` somewhere and it is valid for on prem as for Aura).

While it's fine to have some CDC-reference page in Aura docs containing all these points, we shouldn't regard it as _the place_ where people should discover about these specifics. The UX should be as seamless and "in one place" as possible. The very thorough sysadmins who feel like taking a PhD in CDC before they touch anything-CDC can go through the reference page.

**Comment by l-heemann** [comment]
> it should be in the UI

AFAIK Aura is working on this.

I feel like there is value to having a more lightweight "tutorial-ish" (reference?) page in the Aura docs which can link through to the nitty gritty details in the CDC docs. Any link directly to the CDC docs will likely introduce a lot of noise about issues that the Aura customer doesn't need to think about (such as logRetentionPolicy).

Vice versa we then wouldn't need to have Aura specifics documented in the CDC docs, which are noise to self-hosted customers. I can see the benefits of having Aura specifics included in the CDC docs though. 🤷
